src/improvements/tasks/MultisigTask.sol

```diff
@@ -73,30 +63,31 @@ abstract contract MultisigTask is Test, Script, StateOverrideManager, TaskManage
         public
         returns (VmSafe.AccountAccess[] memory, Action[] memory, bytes32, bytes memory)
     {
-        return simulateRun(taskConfigFilePath, signatures, address(0));
+        return _simulateRun(taskConfigFilePath, signatures, address(0));
     }
 
     /// @notice Runs the task with the given configuration file path.
     function simulateRun(string memory taskConfigFilePath)
         public
         returns (VmSafe.AccountAccess[] memory, Action[] memory, bytes32, bytes memory)
     {
-        return simulateRun(taskConfigFilePath, "", address(0));
+        return _simulateRun(taskConfigFilePath, "", address(0));
     }
 
     /// @notice Executes the task with the given configuration file path and signatures.
     function executeRun(string memory taskConfigFilePath, bytes memory signatures)
         public
         returns (VmSafe.AccountAccess[] memory)
     {
-        _taskSetup(taskConfigFilePath, new address[](0));
-        Action[] memory actions = build();
+        (TaskPayload memory payload, Action[] memory actions) = _taskSetup(taskConfigFilePath, new address[](0));
+        address rootSafe = payload.safes[payload.safes.length - 1];
+        uint256 rootSafeNonce = payload.originalNonces[payload.originalNonces.length - 1];
 
-        (VmSafe.AccountAccess[] memory accountAccesses, bytes32 txHash) = execute(signatures, actions);
+        (VmSafe.AccountAccess[] memory accountAccesses, bytes32 txHash) = execute(signatures, payload);
 
-        // validate all state transitions
-        validate(accountAccesses, actions);
-        print(actions, accountAccesses, false, txHash);
+        validate(accountAccesses, actions, rootSafe, rootSafeNonce);
```

Comment:
We should pass the `payload` to `validate`, and let `validate` decide what it is that it needs from it.